pkg/controllers/workapplier/controller_integration_test.go

```diff
@@ -3952,3 +3952,615 @@ var _ = Describe("report diff", func() {
 		})
 	})
 })
+
+var _ = Describe("switch apply strategies", func() {
+	Context("switch from report diff to CSA", Ordered, func() {
+		workName := fmt.Sprintf(workNameTemplate, "d1")
+		// The environment prepared by the envtest package does not support namespace
+		// deletion; each test case would use a new namespace.
+		nsName := fmt.Sprintf(nsNameTemplate, "d1")
+		deployName := fmt.Sprintf(deployNameTemplate, "d1")
```

Comment:
if the ns is different, then why do we need a different deployment?

Reply:
Will change it -> I used the same suffix just for consistency reasons (and it might help correlation), but I don't have a strong opinion on this